docs(design-guidelines): Adds guidance for new features and expands for current.

edonehoo opened the pull request:
Closes #464 and #468

still need to address 
- response action (no guidance for Share)
- download / copy should grab the message contents? what if there is a source card or other complex objects in the message? Should it download the source?
- Any guidance about what should be read out loud in those cases, too?

patternfly-build commented:
Preview: https://chatbot-pr-chatbot-482.surge.sh

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -57,6 +73,52 @@ A ChatBot can share a link to a [quick start](/extensions/quick-starts) that wil
+1. A thank-you card that confirms a user's response was received. Even if you don't use a feedback form, you should still display the thank-you card, so that users can be confident their rating went through.
+<br />
+<br />
+    <div class="ws-docs-content-img" style="width:75%">
```
Spacing is a little different than what we have in dev (we have a column of space below the avatar):

<img width="861" alt="Screenshot 2025-03-17 at 2 40 52 PM" src="https://github.com/user-attachments/assets/d12eb72a-0aae-41eb-b826-12fba300d3c6" />

edonehoo replied:
I believe this should match now -- Kayla pushed some Figma updates!

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -57,6 +73,52 @@ A ChatBot can share a link to a [quick start](/extensions/quick-starts) that wil
+1. A feedback form that collects more details.
+<br />
+<br />
+    <div class="ws-docs-content-img" style="width:75%">
```
Spacing is a little different around header area; see above. Not sure if we need to fix on dev side or design side.

edonehoo replied:
I believe this should match now -- Kayla pushed some Figma updates!

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -149,16 +211,28 @@ If necessary for brand consistency, you can customize the toggle shape and icon.
+When the conversation history is still loading, display skeleton items:
+
+<div class="ws-docs-content-img">
+![Chat history items loading.](./img/loading-state.svg)
```
Button is "New Chat" here and "New chat" everywhere else. We have "New chat" in the code.

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -149,16 +211,28 @@ If necessary for brand consistency, you can customize the toggle shape and icon.
+If there's an error loading the conversation history, display an error screen with steps for resolving the error:
+
+<div class="ws-docs-content-img">
+![Error state in chat history.](./img/error-state.svg)
```
Text differs from code examples; not sure if this matters.

Placeholder: "Search previous conversations"

> Could not load chat history
To try again, check your connection and reload this page. If the issue persists, contact the support team.

edonehoo replied:
Ah good catch, ty! I do want to match up - I think the default value in Figma was just different, but I'll use this text that's already in the code

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -222,22 +296,36 @@ There are a few display modes that users can choose when interacting with a Chat
 </div>
 
-2. **Docked:** Anchors the ChatBot to the side of the page content. When docked, the ChatBot window is persistent, and cannot be toggled.
+2. **Docked:** Anchors the ChatBot to the side of the page, on top of the page content. When docked, the ChatBot window is persistent, and cannot be toggled. To minimize the ChatBot, users will need to return to overlay mode. 
```
Technically, users can add a close button to the header. We just don't provide examples of this.

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -222,22 +296,36 @@ There are a few display modes that users can choose when interacting with a Chat
 </div>
 
-4. **Embedded:** Places the ChatBot within a product as its own page. An embedded ChatBot could be displayed in the product's navigation menu.
+5. **Inline drawer**: Places the ChatBot to the side of the page, inline with the page content.
```
So this is something I want to be very clear about. (I talked to Nicole about this.) What are actually providing is a chatbot that fits nicely and performs nicely in a drawer. We do not provide the drawer. The drawer is up to the product. It's complicated for us to provide a drawer that will work for everyone in the world.

edonehoo replied:
This is what I changed to: _"A ChatBot that can fit and perform within a drawer. This drawer can look different for each product, but will often be placed to the side of the page, inline with the page content."_ 

I also added a similar snippet beside the inline drawer demo, lmk if you have any concerns with that!

rebeccaalpert replied:
That's perfect I think. Thank you!

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -222,22 +296,36 @@ There are a few display modes that users can choose when interacting with a Chat
+
+### Layouts
+
+ChatBot supports a side-by-side [comparison layout](/patternfly-ai/chatbot/overview/demo#comparing-chatbots), which allows users to compare 2 chats at once. This lets users understand how different models could respond to the same prompt, you can add multiple ChatBots within the same window. 
```
```suggestion
ChatBot supports a side-by-side [comparison layout](/patternfly-ai/chatbot/overview/demo#comparing-chatbots), which allows users to compare 2 chats at once. This lets users understand how different models could respond to the same prompt. You can add multiple ChatBots within the same window. 
```

Maybe a typo? Not sure.

edonehoo replied:
Oh yes definitely something like that 😂 I think I lost my train of thought/didn't finish the sentence lol. ty!

rebeccaalpert reviewed (commented):
Overall this looks great to me - just noticed one typo and there's a couple of things I want to call out re: the drawer and some inconsistency stuff in screenshots. If you guys need me to make a change on the code side, I am more than happy to make changes that we need so these match.

kaylachumley reviewed (approved):
looks good! let me know if you need assistance with updated images in figma. I know the library updates dont always reflect whats on org

edonehoo commented:
In my last commit:
- I removed "Share" from message actions 
- Marked the "Close" button for the user feedback form as optional
- Updated the image for "listening" action to try and showcase the pulsing animation affect
- Added some higher level guidance that was pulled from LS info: 
  - how quick starts should open 
  - guidance for accessing the chatbot 
      - Broke down general options into a floating toggle or a masthead toggle, with some guidance for each 
  - guidance for AI actions that launch the chatbots from within page content (like menu items linked to AI actions)
  - new conversation guidance

rebeccaalpert commented on packages/module/patternfly-docs/content/extensions/chatbot/design-guidelines.md:
```diff
@@ -57,6 +73,53 @@ A ChatBot can share a link to a [quick start](/extensions/quick-starts) that wil
+
+### Message actions 
+
+To allow users to interact with bots messages, utilize message actions. Refer to [the message actions React example](/patternfly-ai/chatbot/messages#message-actions) for an interactive visual.
```
Is this a typo? Otherwise it's fine!

```suggestion
To allow users to interact with bot messages, utilize message actions. Refer to [the message actions React example](/patternfly-ai/chatbot/messages#message-actions) for an interactive visual.
```

rebeccaalpert reviewed (approved):
Just one question about a potential typo. Otherwise this looks awesome. Great work all around.

kaylachumley reviewed (approved):
looks good! ty
